### PR TITLE
Add index to notifications table to optimize query

### PR DESCRIPTION
**Pull request by atsmith813**
## What type of PR is this? (check all applicable)
- [x] Optimization

### TLDR
[This query](https://github.com/thepracticaldev/dev.to/blob/c6434d377b37e0a15e75736428a62522123ef20c/app/services/notifications/update.rb#L17-L26) in `/services/notification/update.rb` called by `Notifications::UpdateWorker` keeps failing with `PG::QueryCanceled: ERROR: canceling statement due to statement timeout` resulting in several dead Sidekiq workers.

Screen Shot 2020-02-05 at 4 02 48 PM

This PR adds an index on the `notifications` table, it looks like an appropriate index was missing.

### Some more details
The query reads:
https://github.com/thepracticaldev/dev.to/blob/c6434d377b37e0a15e75736428a62522123ef20c/app/services/notifications/update.rb#L17-L26

I noticed the indexes on the `notifications` table didn't have an appropriate index. There's not an index for the 3 columns this query is calling for (`notifiable_id`, `notifiable_type`, and `action`). Nor is there an index just for the `action` column.

Screen Shot 2020-02-05 at 4 13 17 PM

I then added a handful (1mm+) `Notification` records locally and re-ran the query before and after adding the index. Looking at `EXPLAIN ANALYZE query` in SQL, I found the performance improved (0.786 ms down to 0.080 ms)...hopefully enough for production 🙏.

I'm curious if this is going down the right path or not. My next hunch was that `ORDER BY` caused by calling `.first`.

## Added to documentation?
- [x] no documentation needed

unsure_gif

### CODE REVIEW

File: db/migrate/20200205225813_add_notifiable_id_notifiable_type_action_index_to_notifications.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class AddNotifiableIdNotifiableTypeActionIndexToNotifications < ActiveRecord::Migration[5.2]
+  disable_ddl_transaction!
+
+  def change
+    add_index :notifications, %i[notifiable_id notifiable_type action],
+              unique: false,
+              name: "index_notifications_on_notifiable_id_notifiable_type_and_action",
+              algorithm: :concurrently
```
Comment (citizen428) [inline db/migrate/20200205225813_add_notifiable_id_notifiable_type_action_index_to_notifications.rb]: 👏 for remembering to add `:concurrently`, came here to suggest that only to find it's already there 👍

**Comment by citizen428** [comment]
> My next hunch was that ORDER BY caused by calling `.first`

`first` in this case adds `ORDER BY "notifications"."id" ASC`, which shouldn't be a problem as we have an index for the PK.
